### ISSUE TITLE
Feature Request - replace librespot with go-librespot as it better implements Spotify connect and is currently maintained

### ISSUE THREAD

**Issue by ironcrafter54** [firefrei/docker-snapcast#9]
here is the go-librespot repo if you are curious https://github.com/devgianlu/go-librespot, btw thanks so much for making this I have found it super useful.

**Comment by firefrei** [comment]
Thank you for contributing your idea!
However, I do not see the reason why libraries should be switched. We use the "officially" recommended implementation by Snapcast [1], where the integration is tested and the past proved that it is working without any major issues. Where do you see benefits by using go-librespot?
Furthermore, it doesn't seem to me like librespot isn't maintained anymore. The last commit is from two months ago. We build librespot from scratch, so their released versions on Github do not matter. I just checked the librespot version that the snapcast container in my home is running: it is librespot 0.5.0-dev 8f9bec2, committed in May of this year. The latest dev-image build should already be on a newer version.

[1] https://github.com/badaix/snapcast/blob/develop/doc/player_setup.md#spotify

**Comment by ironcrafter54** [comment]
Some advantages the go-librespot offers is an api that you can interact with, it also reports what is playing to spotify which librespot does not currently due as far as I am aware of. Also some of the podcast I try to listen to do not play through the current implementation of librespot, for example the vergecast just does not play for some reason while there are no issues playing the podcast darknet diaries, I believe, but am not sure, the switching to go-librespot would fix these issues. Just thought Id'e share the upsides I see, thanks for creating this project btw its made getting into using snapcast so much easier.

**Comment by firefrei** [comment]
I just tried playing the vergecast in my home via librespot and it worked without any issue. It seems to be a "local" problem with your setup (or geo location?). Also the song title is displayed in snapweb.
Additionally, I checked the go-librespot repository and, unfortunately, I cannot find a proper configuration documentation (except the API doc) that would explain how to configure the library to output the samples to stdout, so it can be connected to snapcast's "pipe" input. It seems like it can "only" play directly to alsa. How song information could be provided to snapcast remains also unclear.  
In summary, I do not see a need to change to another library at the moment. Thank you for sharing the link to an alternative implementation, I do appreciate your contribution :-)  I will check it again once librespot is discontinued, faces major issues, or go-librespot has significant better integration support with snapcast.
